### PR TITLE
test: 23424 fix flaky e2e test for change language

### LINKED ISSUE #23424: Fix "Settings - general tab, validate the change language..." flaky tests
### What is this about?

image

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/72921/workflows/bd1e2089-b400-4044-b9b2-816e83a42fba/jobs/2472531

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by hjetpoluru**
## **Description**

This PR fixes flaky test for change language and converted the test spec to typescript

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23505?quickstart=1)

The actual flaky test scenario is: "User selects 'Español' from the dropdown, changes it back to English, and verifies that the word has correctly changed back to English."

The test scenario involved changing the language from English to Spanish and then back from Spanish to English.

However, when I clicked on the dropdown to change the language from English to Spanish, the options did not collapse.
Consequently, my attempt to change from Spanish back to English failed. I have now included a page refresh in the process and changed the language back to English.

image

## **Related issues**

Fixes: #23424 

## **Manual testing steps**

Run the test in codespace or locally --> Checkout to the branch
yarn install
yarn start:test
yarn test:e2e:single test/e2e/tests/settings/change-language.spec.ts --browser=chrome --debug --leave-running
yarn test:e2e:single test/e2e/tests/settings/change-language.spec.ts --browser=firefox --debug --leave-running

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/settings/change-language.spec.ts [hunk 1]
```diff
@@ -30,74 +28,50 @@ const selectors = {
...
         await dropdownElement.click();
-        const options = await dropdownElement.findElements(By.css('option'));
+        const options = await dropdownElement.findElements({ css: 'option' });
         await options[languageIndex].click();
 
-        await driver.assertElementNotPresent('.loading-overlay__spinner', {
-          waitAtLeastGuard: 100,
-        });
-
         const islabelTextChanged = await driver.isElementPresent(
```
Comment (HowardBraham) [inline test/e2e/tests/settings/change-language.spec.ts:75]: The new spell checker that David Murdoch installed noticed that the capitalization on this should be `isLabelTextChanged`

File: test/e2e/tests/settings/change-language.spec.ts [hunk 2]
```diff
@@ -30,74 +28,50 @@ const selectors = {
...
-          selectors.primaryCurrencyLabel,
-        );
-        assert.equal(isLanguageLabelChanged, true, 'Language did not change');
-
-        await driver.assertElementNotPresent('.loading-overlay__spinner', {
-          waitAtLeastGuard: 100,
-        });
+        await driver.refresh();
 
+        // Change back to english and verify that the word is correctly changed back to english
```
Comment (HowardBraham) [inline test/e2e/tests/settings/change-language.spec.ts:65]: Language names like `English` and `Spanish` should always be capitalized

**Review by HowardBraham — COMMENTED** [review]
Just some nits really, but I do want to know what was the change to make it less flaky?

File: test/e2e/tests/settings/change-language.spec.ts (hunk 1 above)
Comment (hjetpoluru) [reply test/e2e/tests/settings/change-language.spec.ts]: Good catch! It's difficult to spot with the human eye.

File: test/e2e/tests/settings/change-language.spec.ts (hunk 2 above)
Comment (hjetpoluru) [reply test/e2e/tests/settings/change-language.spec.ts]: Ok done.

**Comment by hjetpoluru** [comment]
@HowardBraham, I really appreciate your keen attention to detail. 

The actual flaky test scenario is: "User selects 'Español' from the dropdown, changes it back to English, and verifies that the word has correctly changed back to English."

The test scenario involved changing the language from English to Spanish and then back from Spanish to English.

However, when I clicked on the dropdown to change the language from English to Spanish, the options did not collapse. 
Consequently, my attempt to change from Spanish back to English failed. I have now included a page refresh in the process and changed the language back to English.
Screenshot 2024-03-15 at 10 28 20 AM

The last thing I did was combine some of the test scenarios and convert them to TypeScript, which is why the changes I made were not noticeable.

I have a question for you: I have configured the spell check settings, but it did not highlight any spelling errors.

**Review by HowardBraham — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]

**Pull request merged**